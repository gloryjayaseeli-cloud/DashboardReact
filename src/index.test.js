import React from 'react';
jest.mock('./App', () => () => <div data-testid="mock-app">Mock App</div>);
jest.mock('./reportWebVitals', () => jest.fn());
jest.mock('react-dom/client'); 
jest.mock('./store/index', () => ({
  store: {},
  persistor: {},
}));

describe('Application Root', () => {
  let ReactDOM;
  let mockRender;

  beforeEach(() => {
   jest.resetModules();

    ReactDOM = require('react-dom/client');
    mockRender = jest.fn();

    ReactDOM.createRoot.mockReturnValue({
      render: mockRender,
    });
  });

  it('should render the App with all the required providers in the correct order', () => {
    const rootElement = document.createElement('div');
    rootElement.id = 'root';
    document.body.appendChild(rootElement);
   require('./index.js');

     expect(ReactDOM.createRoot).toHaveBeenCalledWith(rootElement);

     expect(mockRender).toHaveBeenCalledTimes(1);

     const renderedComponentTree = mockRender.mock.calls[0][0];
    expect(renderedComponentTree).toMatchSnapshot();
  });
});

