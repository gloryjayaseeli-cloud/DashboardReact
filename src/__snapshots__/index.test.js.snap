// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Application Root should render the App with all the required providers in the correct order 1`] = `
Object {
  "$$typeof": Symbol(react.transitional.element),
  "_owner": null,
  "_store": Object {},
  "key": null,
  "props": Object {
    "children": Object {
      "$$typeof": Symbol(react.transitional.element),
      "_owner": null,
      "_store": Object {},
      "key": null,
      "props": Object {
        "children": Object {
          "$$typeof": Symbol(react.transitional.element),
          "_owner": null,
          "_store": Object {},
          "key": null,
          "props": Object {
            "children": Object {
              "$$typeof": Symbol(react.transitional.element),
              "_owner": null,
              "_store": Object {},
              "key": null,
              "props": Object {
                "children": Object {
                  "$$typeof": Symbol(react.transitional.element),
                  "_owner": null,
                  "_store": Object {},
                  "key": null,
                  "props": Object {},
                  "type": [Function],
                },
              },
              "type": [Function],
            },
            "loading": null,
            "persistor": Object {},
          },
          "type": [Function],
        },
        "store": Object {},
      },
      "type": [Function],
    },
  },
  "type": Symbol(react.strict_mode),
}
`;
